File: app/camera/ios/DocumentScanner.m
#import "DocumentScanner.h"

@implementation DocumentScanner

RCT_EXPORT_MODULE();

- (NSArray<NSString *> *)supportedEvents
{
  return @[@"Scan"];
}

RCT_EXPORT_METHOD(open)
{
  [self sendEventWithName:@"Scan" body:@{@"name": @"unicorn test"}];
}

@end

File: app/camera/ios/DocumentScanner.swift
import Foundation

@objc(DocumentScanner)
class DocumentScanner: RCTEventEmitter {
  @objc
  override func supportedEvents() -> [String] {
    return [
      "Scan"
    ]
  }

  @objc(open)
  func open() -> Void {
    NSLog("Hello from swift");
    self.sendEvent(
      withName: "Scan",
      body: "a scan event from swift"
    )
  }
}
